src/diff/index.js

```diff
@@ -376,3 +385,34 @@ function catchErrorInComponent(error, component) {
 	}
 	throw error;
 }
+
+/**
+ * Determine which currently attached DOM node to use when beginning to
+ * diff the children of a VNode
+ * @param {import('../internal').VNode} oldVNode The old VNode whose children
+ * are about to be diffed
+ * @param {import('../internal').PreactElement[]} excessDomChildren If hydrating,
+ * the currently attached DOM elements that are being hydrated
+ * @returns {import('../internal').PreactElement | Text | undefined}
+ */
+export function getFirstChildDom(oldVNode, excessDomChildren) {
```

Comment:
Looking to inline this function to save some bytes

Reply:
I had a go at this and while it should be possible I'm having a hard time doing that. It breaks all kinds of tests for me.

Reply:
Lol same! That's where I'm at too. Haven't looked at it in a day or two so maybe my fresh eyes will see what I'm missing now

Reply:
Got it working, but it's kinda weird. I'll open it as a separate PR for discussion after we merge this one.